fix PULSEDEV-N/A Dockerfile: Fix image build by updating CA certificates

mlobofeedzai opened the pull request:
It was no longer working because certificates from old ubuntu version were no longer valid.

MLobo1997 committed:
fix PULSEDEV-N/A Dockerfile: Fix image build.

It was no longer working because certificates from old ubuntu version were no longer valid.

MLobo1997 committed:
Revert "fix PULSEDEV-N/A Dockerfile: Fix image build."

This reverts commit cab80a03ca0dbc9690dedd782cd36b085240f1da.

MLobo1997 committed:
fix PULSEDEV-N/A Dockerfile: Fix image build.

It was no longer working because certificates from old ubuntu version were no longer valid. So, we now update the CA certificates.

AlbertoEAF commented:
The comment is not entirely accurate yet, I suggest changing to:

This image uses very specific (and old) versions of the OS and tools. The aim is two-fold. First, to match the build pipeline used for the official Microsoft LightGBM  releases, for which there are official tests, avoiding unexpected bugs. Second, retain libc==2.12, to ensure compatibility of the produced LightGBM binaries with any target machine.

MLobo1997 committed:
fix PULSEDEV-N/A Dockerfile: Improve Dockerfile comment.

So that message is clearer.

shengwangsw commented on docker/lightgbm-ci-build-env/Dockerfile:
```diff
@@ -60,8 +64,22 @@ RUN apt-get update \
+# Update ca certificates to safely perform curls
+RUN mkdir -p ~/.ca-certificates-src \
+    && cd ~/.ca-certificates-src \
+    && wget https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/ca-certificates/20210119~20.04.2/ca-certificates_20210119~20.04.2.tar.xz \
```
for security reason, consider downloading the file.
the red flag is that you are executing `sudo make and sudo make install`

mlobofeedzai replied:
Great point. fixed

MLobo1997 committed:
fix PULSEDEV-N/A Dockerfile: Include CA certificates in repo instead of WGETing

This was a security alert made by @shenggwang. Automating wget followed by a `sudo make` was a risky vulnerability.

shengwangsw reviewed (approved):
LGTM
